src/table/body-cell/td-selection-element.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import clsx from 'clsx';
+import React, { KeyboardEventHandler } from 'react';
+import SelectionControl from '../selection-control/';
+import { TableProps } from '../interfaces.js';
+import { TableTdElementProps } from './td-element.js';
+import { useVisualRefresh } from '../../internal/hooks/use-visual-mode';
+import styles from './styles.css.js';
+
+type TableBodySelectionCellProps = Omit<
+  TableTdElementProps,
+  'style' | 'nativeAttributes' | 'onClick' | 'tdRef' | 'hasSelection' | 'wrapLines' | 'columnId'
+> & {
+  selectionType?: 'single' | 'multi';
+  onFocusUp?: KeyboardEventHandler;
+  onFocusDown?: KeyboardEventHandler;
+  onShiftToggle: (value: boolean) => void;
+  itemSelectionProps: {
```

Comment:
Would it make sense to define a separate interface for itemSelectionProps? Possibly inside `use-selection.ts`